### DISCUSSION TITLE
How tsup can bundle .d.ts?

### DISCUSSION THREAD

**Discussion by rqbazan** [egoist/tsup#795] [Q&A]
Hi there,

I'm really interested about how tsup is so fast bundling declaration files. I was trying to use some alternative because I'm using Vite for building a library and [dts-bundle-generator](https://github.com/timocov/dts-bundle-generator) is so slow for me.

I'd like to have enough knowledge (and time) to build something like `dts-bundle-generator` but faster.

**Comment by theaungmyatmoe** [comment]
Declaration files didn't need to be bundled. They are just an annotations for type hinting in development environment. It is nothing when you deploy for the browser. You can use `--dts` to generate `.d.ts` files.

**Reply by pofulu** [reply]
I believe there're some use cases for bundle `.d.ts` files. For example, build a library for non-browser/node environments:
- https://github.com/Tencent/puerts
- https://sparkar.facebook.com/ar-studio/

**Reply by theaungmyatmoe** [reply]
I know that. You are talking about generation declaration files. That feature is including with this bundler. It isn't about bundling, just generation declaration files. `--dts` to generate such declaration file with this bundler.
